avm/res/network/firewall-policy/main.bicep

```
@@ -0,0 +1,212 @@
+
+@description('Optional. Resource ID of the base policy.')
+param basePolicyResourceId string?
+
+@description('Optional. Enable DNS Proxy on Firewalls attached to the Firewall Policy.')
+param enableProxy bool = false
+
+@description('Optional. List of Custom DNS Servers.')
+param servers array?
+
+@description('Optional. A flag to indicate if the insights are enabled on the policy.')
+param insightsIsEnabled bool = false
+
+@description('Optional. Default Log Analytics Resource ID for Firewall Policy Insights.')
+param defaultWorkspaceId string?
+
+@description('Optional. List of workspaces for Firewall Policy Insights.')
+param workspaces array?
+
+@description('Optional. Number of days the insights should be enabled on the policy.')
+param retentionDays int = 365
+
+@description('Optional. List of rules for traffic to bypass.')
+param bypassTrafficSettings array?
+
+@description('Optional. List of specific signatures states.')
+param signatureOverrides array?
+
+@description('Optional. The configuring of intrusion detection.')
+@allowed([
+  'Alert'
+  'Deny'
+  'Off'
+])
+param mode string = 'Off'
+
+@description('Optional. Tier of Firewall Policy.')
+@allowed([
+  'Premium'
+  'Standard'
+])
+param tier string = 'Standard'
+
+@description('Optional. List of private IP addresses/IP address ranges to not be SNAT.')
+param privateRanges array = []
+
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+@description('Optional. The operation mode for automatically learning private ranges to not be SNAT.')
+param autoLearnPrivateRanges string = 'Disabled'
+
+@description('Optional. The operation mode for Threat Intel.')
+@allowed([
+  'Alert'
+  'Deny'
+  'Off'
+])
+param threatIntelMode string = 'Off'
+
+@description('Optional. A flag to indicate if SQL Redirect traffic filtering is enabled. Turning on the flag requires no rule using port 11000-11999.')
+param allowSqlRedirect bool = false
+
+@description('Optional. List of FQDNs for the ThreatIntel Allowlist.')
+param fqdns array?
+
+@description('Optional. List of IP addresses for the ThreatIntel Allowlist.')
+param ipAddresses array?
+
+@description('Optional. Secret ID of (base-64 encoded unencrypted PFX) Secret or Certificate object stored in KeyVault.')
+#disable-next-line secure-secrets-in-params // Not a secret
+param keyVaultSecretId string?
+
+@description('Optional. Name of the CA certificate.')
+param certificateName string?
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. Rule collection groups.')
+param ruleCollectionGroups array?
+
+var formattedUserAssignedIdentities = reduce(map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }), {}, (cur, next) => union(cur, next)) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
+
+var identity = !empty(managedIdentities) ? {
+  type: !empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'UserAssigned' : null
```

```suggestion
  type: 'UserAssigned'
```

I guess, this will be good enough